src/functions/Output.ps1

```diff
@@ -670,6 +747,10 @@ function Get-WriteScreenPlugin ($Verbosity) {
             throw "Unsupported CI format '$($PesterPreference.Output.CIFormat.Value)'"
         }
 
+        if ($PesterPreference.Run.SkipRemainingOnFailure.Value -notin 'None', 'Block', 'Container', 'Run') {
+            throw "Unsupported SkipRemainingOnFailure option '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
+        }
+
```

Comment:
Yeah since `Get-WriteScreenPlugin` throws this error inside `EachTestTearDown` for the output options, I might do the same for my plugin.

Reply:
Yeah wanted to do this but it only seems to work when put in `Get-WriteScreenPlugin`? If I do it in my own plugin in `RunStart` it shows a huge stack trace:

```pwsh
if ($PesterPreference.Run.SkipRemainingOnFailure.Value -notin 'None', 'Block', 'Container', 'Run') {
    $p.RunStart = {
        throw "Unsupported SkipRemainingOnFailure option '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
    }
}
```

```pwsh
Invoking step RunStart failed:
Result :Error :
Exception: C:\Users\Armaan\Documents\github-repos\Pester\src\functions\Get-SkipRemainingOnFailurePlugin.ps1:8:13
Line |
   8 |              throw "Unsupported SkipRemainingOnFailure option '$($Pest …
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Unsupported SkipRemainingOnFailure option 'Non'

at <ScriptBlock>, C:\Users\Armaan\Documents\github-repos\Pester\src\functions\Get-SkipRemainingOnFailurePlugin.ps1: line 8
at Invoke-PluginStep, C:\Users\Armaan\Documents\github-repos\Pester\src\Pester.Runtime.ps1: line 1243
at Run-Test, C:\Users\Armaan\Documents\github-repos\Pester\src\Pester.Runtime.ps1: line 1043
at Invoke-Test, C:\Users\Armaan\Documents\github-repos\Pester\src\Pester.Runtime.ps1: line 1950
at Invoke-Pester<End>, C:\Users\Armaan\Documents\github-repos\Pester\src\Main.ps1: line 1059
at <ScriptBlock>, <No file>: line 1
```

Reply:
This doesn't happen though when the option checking is in `Get-WriteScreenPlugin` like what I've already done. Perhaps the ordering of plugins if affecting this?

Reply:
What I could do is just throw the error inside `Invoke-Pester` like so when adding the plugin:

```pwsh
if ($PesterPreference.Run.SkipRemainingOnFailure.Value -notin 'None', 'Block', 'Container', 'Run') {
    throw "Unsupported SkipRemainingOnFailure option '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'"
}
else {
    $plugins += @(Get-SkipRemainingOnFailurePlugin)
}
```

Reply:
I see. It's not about the plugin, but the plugin-step being invoked when you throw. The setup/teardown-steps are called using `Invoke-ScriptBlock` which has a different error-handling logic than the general steps like RunStart which are called with `Invoke-PluginStep`. The latter re-throws the exception.

If you use `$p.EachTestTeardownEnd = { throw 'demo' }` in your plugin it should look the same. At that point, we've already executed too much though. If we're gonna throw, we might as well do it prior to discovery etc. to avoid wasting the user's time. I'm not sure why the Output-validation is done so late tbh. Do you remember @nohwnd ?

If you validate in `Main.ps1` or as part of `Get-SkipRemainingOnFailurePlugin`  (to keep it isolated) then you can throw when the plugin is generated. It's not a single-line error, but simple enough. Ex.

```powershell
# code
function Get-SkipRemainingOnFailurePlugin {
    ....
    elseif ($PesterPreference.Run.SkipRemainingOnFailure.Value -notin 'None', 'Block', 'Container', 'Run') {
        $p.EachTestSetupStart = { throw "Unsupported Run.SkipRemainingOnFailure option '$($PesterPreference.Run.SkipRemainingOnFailure.Value)'" }
    }

    New-PluginObject @p
}

# demo
PS /workspaces/Pester/Samples> /workspaces/Pester/Samples/demoSkipRemaining.ps1
System.Management.Automation.RuntimeException: Unsupported Run.SkipRemainingOnFailure option 'Blocks'
at Get-SkipRemainingOnFailurePlugin, /workspaces/Pester/src/functions/Get-SkipRemainingOnFailurePlugin.ps1: line 76
at Invoke-Pester<End>, /workspaces/Pester/src/Main.ps1: line 966
at <ScriptBlock>, /workspaces/Pester/Samples/demoSkipRemaining.ps1: line 38
```

Reply:
Thanks for this, really insightful 🙂. After stepping through the code I ran into similar behaviour, which makes me understand more how the setup/teardown steps work in the background. 

For now I'm just handling this in `Main.ps1`, which will just throw the exception like what you've shown.